<launch>
  <machine name="local" address="localhost" default="true" />

  <param name="use_sim_time" value="true"/>
  <arg name="dataset" default="" />
  <arg name="rate" default="1.0" />
  <arg name="pose_output" default="trajectory_tiny_res.txt" />

  <node name="tinySLAM" pkg="slam_fmwk" type="tiny_slam"
        cwd="node" required="true" output="screen">
    <remap from="laser_scan" to="/base_scan" />

    <!-- Modules setup -->
    <param name="slam/cell/type" value="avg" />
    <param name="slam/occupancy_estimator/type" value="const" />
    <param name="ros/skip_exceeding_lsr_vals" value="false" />

    <!-- Map setup-->
    <param name="slam/map/height_in_meters" value="100" />
    <param name="slam/map/width_in_meters" value="100" />
    <param name="slam/map/meters_per_cell" value="0.1" />

    <!-- Internal constants setup -->
    <param name="slam/cell/base_occupied_prob" value="0.95" />
    <param name="slam/cell/base_empty_prob" value="0.01" />

    <!-- Internal scan matcher constants-->
    <param name="slam/scmtch/MC/sigma_XY" value="0.2" />
    <param name="slam/scmtch/MC/sigma_theta" value="0.1" />
    <param name="slam/scmtch/MC/limit_of_bad_attempts" value="20" />
    <param name="slam/scmtch/MC/limit_of_total_attempts" value="100" />

    <!-- Internal constants for ros-->
    <param name="ros/rviz/map_publishing_rate" value="5.0" />
    <param name="ros/tf/buffer_duration" value="5.0" /> <!-- in seconds-->
    <param name="ros/filter_queue_size" value="1000" />
    <param name="ros/subscribers_queue_size" value="1000" />
  </node>

  <node name="bag_adapter" pkg="slam_fmwk"
        type="wg_pr2_bag_adapter" required="true" output="screen">
    <param name="tf_map_frame_id" value="map" />
    <param name="tf_odom_frame_id" value="odom_combined" />

    <remap from="provided_grid_orig" to="/map_bag" />
    <remap from="provided_grid" to="/ground_truth_map" />

    <remap from="tf_in" to="/tf_bag" />
    <remap from="tf_out" to="/tf" />
  </node>
  
  <node name="player" pkg="rosbag" type="play"
        args="--delay=3 --clock --rate $(arg rate) $(arg path)" />

  <node name="rviz" pkg="rviz" type="rviz" respawn="true"
        args="-d $(find slam_fmwk)/rviz/debug.rviz" />
        
  <node name="PoseLogger" pkg="slam_fmwk" type="pose_logger.py" cwd="node" required="true">
      <param name="base_frame" value="base_link" />
      <param name="target_frame" value="map" />
      <param name="output_file" value="$(arg pose_output)" />
  </node>

</launch>